### PR TITLE
Quick lookup for cached urls on the shelf

### PR DESCRIPTION
**Pull request by bmorris3**
This is a quick convenience function I wrote today on an airplane (without internet), when I was trying to figure out which URLs I had cached. It returns a list of the URLs that have been stored in the cache.

### CODE REVIEW

**Comment by bsipocz** [comment]
@bmorris3 - This will need a test and a changelog. I'm not sure which version though, can be 2.0 if @eteq is happy with it, otherwise 3.0 (then you need to wait until tomorrow when the branching is done and the new sections are added).

File: astropy/utils/data.py [hunk 1]
```diff
@@ -1370,3 +1370,27 @@ def _release_download_cache_lock():
         msg = 'Error releasing lock. "{0}" either does not exist or is not ' +\
               'a directory.'
         raise RuntimeError(msg.format(lockdir))
+
+
+def _urls_on_shelf():
```
Comment (eteq) [inline astropy/utils/data.py]: Any reason for it to be private?  I don't see any reason not to make it public, actually! (i.e., no leading underscore)

File: astropy/utils/data.py [hunk 2]
```diff
@@ -1370,3 +1370,27 @@ def _release_download_cache_lock():
...
+    """
+    Get the list of URLs in the shelf. Useful for looking up what files are
```
Comment (eteq) [inline astropy/utils/data.py]: "on the shelf" is an implementation detail - maybe "in the cache"?

File: astropy/utils/data.py (hunk 1 above)
Comment (eteq) [inline astropy/utils/data.py]: And related to my comment in the line just below: if it is public, maybe the name should be `get_cached_urls`?

File: astropy/utils/data.py [hunk 3]
```diff
@@ -1370,3 +1370,27 @@ def _release_download_cache_lock():
...
+    cached when you don't have internet access.
+
+
+    Returns
+    -------
+    in_cache : bool
```
Comment (eteq) [inline astropy/utils/data.py]: This looks like a copy-and-paste typo?  Shouldn't it be the list of URLs that are cached?

File: astropy/utils/data.py (hunk 1 above)
Comment (bmorris3) [inline astropy/utils/data.py]: Perfect. I was thinking too narrowly about who would care, but I suppose so many affiliated packages use the download utils that this is probably useful for lots of people.

File: astropy/utils/data.py (hunk 2 above)
Comment (bmorris3) [inline astropy/utils/data.py]: I got carried away by the analogy.

File: CHANGES.rst [hunk 4]
```diff
@@ -213,6 +213,9 @@ New Features
   - Added a new ``dataurl_mirror`` configuration item in ``astropy.utils.data``
     that is used to indicate a mirror for the astropy data server. [#5547]
 
+  - Added a new convenience method ``get_cached_urls`` to ``astropy.utils.data``
+    for getting a list of the URLs in your cache.
```
Comment (eteq) [inline CHANGES.rst]: Needs the PR number at the end.  But that's trivial so I'll just fix it myself.

**Comment by eteq** [comment]
LGTM!  Ready to merge once the tests pass.

**Comment by eteq** [comment]
Actually, Travis appears to have already passed, except for some remote-data related errors that seem to be all timeouts and hence are clearly not relevant.  So merging now!

**Pull request merged**